### PR TITLE
Add some content to FAQ, do some formatting.

### PR DESCRIPTION
**Pull request by zach-klippenstein**

### CODE REVIEW

File: docs/src/markdown/documentation/faq.md [hunk 1]
```diff
@@ -9,14 +9,21 @@ navigation:
 # Frequently Asked Questions
 
 
+### Isn't this basically React/Elm?
+
+[React](https://reactjs.org/) and [the Elm architecture](https://guide.elm-lang.org/architecture/) were both strong influences for this library. However both those libraries are both written for JavaScript. Workflows are written in and for both Kotlin and Swift, making use of features of those languages, and with usability from those languages as a major design goal. There are also a few architectural differences:
```
Comment (loganj) [inline docs/src/markdown/documentation/faq.md]: "both those libraries are both"

**Comment by zach-klippenstein** [comment]
I'm cheating and using this site to generate the tables: http://www.tablesgenerator.com/markdown_tables#

**Comment by zach-klippenstein** [comment]
I think I'd be ok merging this with the TK's in place, we can flesh out the table more later.

**Comment by zach-klippenstein** [comment]
Swift WorkflowUI shard is erroring out, filed as #201.

**Comment by rjrjr** [comment]
What's this thing at the top of the file?

```
---
title: FAQ	title: FAQ
index: 100	index: 100
navigation:	navigation:
    visible: true	    visible: true
    path: documentation	    path: documentation
---	---
```

File: docs/src/markdown/documentation/faq.md [hunk 2]
```diff
@@ -9,14 +9,31 @@ navigation:
...
+
+|  | React | Elm | Workflow |
+|---|---|---|---|
+| **Modularity** | `Component` | TK | `Workflow` is analogous to React's `Component` |
+| **State** | Each `Component` has a `state` property that is read directly and updated via a `setState` method. | State is called `Model` in Elm. | `Workflow`s have an associated state type. The state can only be updated when the input changes, or with a `WorkflowAction`. |
+| **Views** | `Component`s have a `render` method that returns a tree of elements. | Elm applications have a `view` function that returns a tree of elements. | Since workflows are not tied to any particular UI view layer, they can have an arbitrary rendering type. The `compose()` method returns this type. |
+| **Dependencies** | React allows parent components to pass "props" down to their children. | TK | In Swift, `Workflow`s are often structs that need to be initialized with their dependencies and configuration data from their parent. In Kotlin, they have a separate associated generic type that is always passed down from the parent. `Workflow` instances can also inject dependencies, and play nicely with dependency injection frameworks.
```
Comment (rjrjr) [inline docs/src/markdown/documentation/faq.md]: In Kotlin, they have an associated parameter type (`InputT`) that is…

File: docs/src/markdown/documentation/faq.md [hunk 3]
```diff
@@ -9,14 +9,31 @@ navigation:
 # Frequently Asked Questions
 
 
+### Isn't this basically React/Elm?
+
+[React](https://reactjs.org/) and [the Elm architecture](https://guide.elm-lang.org/architecture/) were both strong influences for this library. However both those libraries are written for JavaScript. Workflows are written in and for both Kotlin and Swift, making use of features of those languages, and with usability from those languages as a major design goal. There are also a few architectural differences:
+
+|  | React | Elm | Workflow |
+|---|---|---|---|
+| **Modularity** | `Component` | TK | `Workflow` is analogous to React's `Component` |
```
Comment (rjrjr) [inline docs/src/markdown/documentation/faq.md]: Is `TK` how we're saying `TBD` these days?

Comment (zach-klippenstein) [reply docs/src/markdown/documentation/faq.md]: https://en.wikipedia.org/wiki/To_come_(publishing)

**Pull request merged**